torch/_dynamo/variables/builder.py

```diff
@@ -521,6 +521,9 @@ def index_source(key):
                 source=self.source,
                 guards=make_guards(GuardBuilder.ID_MATCH),
             )
+        elif isinstance(value, np.generic):
+            # numpy array scalars: convert to 0D arrays
+            return self.wrap_numpy_ndarray(np.asarray(value))
```

Comment:
Actually, I think a better fix is to put this here:
https://github.com/pytorch/pytorch/blob/8a8e9a6c5f5b6103e7e877b8c0b27aca7b8323db/torch/_dynamo/variables/builder.py#L306
and then teach `wrap_numpy_ndarray` to handle np.generics.

Reply:
better is not the word. it's just that it'd be together with its friend np.ndarray

Reply:
as is it's next to `elif is_numpy()` (next line)